ignite/metrics/precision.py

```diff
@@ -1,9 +1,10 @@
 from __future__ import division
+import warnings
 
 import torch
 
-from .metric import Metric
-from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric
+from ignite.exceptions import NotComputableError, UndefinedMetricWarning
```

Comment:
Do we need to define a custom warning? Could/Should we not just use the generic RuntimeWarning?

Reply:
I think the argument here is the same as any custom Exception/Warning. This allows more granular control for the user. They can handle just this specific warning. That being said, I'm fine with using RuntimeWarning if you'd prefer.

Reply:
Well I guess if we remove the warning or swap it with a logging line then this is a moot point